pkg/types/vector.go

```diff
@@ -131,6 +131,23 @@ func (v VectorFloat32) EstimatedMemUsage() int {
 	return int(unsafe.Sizeof(v)) + len(v.data)
 }
 
+// PeekBytesAsVectorFloat32 trys to peek some bytes from b, until
+// we can deserialize a VectorFloat32 from those bytes.
+func PeekBytesAsVectorFloat32(b []byte) (n int, err error) {
+	if len(b) < 4 {
+		err = errors.Errorf("bad VectorFloat32 value header (len=%d)", len(b))
+		return
+	}
+
+	elements := binary.LittleEndian.Uint32(b)
+	totalDataSize := elements*4 + 4
+	if len(b) < int(totalDataSize) {
+		err = errors.Errorf("bad VectorFloat32 value (len=%d, expected=%d)", len(b), totalDataSize)
+		return
+	}
+	return int(totalDataSize), nil
+}
+
```

Comment:
In CSE PR `ZeroCopyDeserializeVectorFloat32` is re-implemented by calling `PeekBytesAsVectorFloat32`.

Reply:
modified.